lib/custom/src/MShop/Service/Provider/Payment/Stripe.php

```diff
@@ -33,20 +111,225 @@ protected function getConfigPrefix()
+	 * @return \Aimeos\MShop\Common\Item\Helper\Form\Standard Form object with URL, action and parameters to redirect to
+	 *    (e.g. to an external server of the payment provider or to a local success page)
+	 */
+	public function process(\Aimeos\MShop\Order\Item\Iface $order, array $params = [])
+	{
+		if( !isset( $params['paymenttoken'] ) ) {
+			return $this->getPaymentForm( $order, $params );
+		}
+		return $this->processOrder($order, $params);
+	}
+
+
+	/**
+	 * Returns the payment form for entering payment details at the shop site.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Iface $order Order object
+	 * @param array $params Request parameter if available
+	 * @return \Aimeos\MShop\Common\Item\Helper\Form\Iface Form helper object
+	 */
+	protected function getPaymentForm(\Aimeos\MShop\Order\Item\Iface $order, array $params)
+	{
+		$list = [];
+		$feConfig = $this->feConfig;
+
+		foreach ($feConfig as $key => $config) {
+			$list[$key] = new \Aimeos\MW\Criteria\Attribute\Standard($config);
+		}
+		$url = $this->getConfigValue(array('payment.url-self'));
+		return new \Aimeos\MShop\Common\Item\Helper\Form\Standard($url, 'POST', $list, false, $this->getHtmlForm());
+	}
+
+
+
+
+	/**
+	 * Returns the configuration attribute definitions of the provider to generate a list of available fields and
+	 * rules for the value of each field in the frontend.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Base\Iface $basket Basket object
+	 * @return array List of attribute definitions implementing \Aimeos\MW\Common\Critera\Attribute\Iface
+	 */
+	public function getConfigFE(\Aimeos\MShop\Order\Item\Base\Iface $basket)
+	{
+		return [];
+	}
+
+	/**
+	 * Returns the data passed to the Omnipay library
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Base\Iface $base Basket object
+	 * @param $orderid Unique order ID
+	 * @param array $params Request parameter if available
+	 */
+	protected function getData( \Aimeos\MShop\Order\Item\Base\Iface $base, $orderid, array $params )
+	{
+		$data = parent::getData($base,$orderid,$params);
+		if( isset($params['paymenttoken']) ){
+			$data['token'] = $params['paymenttoken'];
 		}
+		return $data;
+	}
+
+	/**
+	 * Checks the frontend configuration attributes for validity.
+	 *
+	 * @param array $attributes Attributes entered by the customer during the checkout process
+	 * @return array An array with the attribute keys as key and an error message as values for all attributes that are
+	 *    known by the provider but aren't valid resp. null for attributes whose values are OK
+	 */
+	public function checkConfigFE(array $attributes)
+	{
+		return [];
+		//return $this->checkConfig($this->feConfig, $attributes);
+	}
+
+	public function getHtmlForm()
+	{
+		return '<script src="https://js.stripe.com/v3/"></script>
+		<script type="text/javascript">
+    	$(document).ready(function () {
+        var stripe = Stripe("'.$this->getConfigValue( array( $this->getConfigPrefix() . '.publishableKey' ), '' ).'");
```

Comment:
Can you put everything in an JS object so it doesn't pollute the global JS name space?
Furthermore you should remove all code that is inactive and all debug statements.